frappe/model/base_document.py

```diff
@@ -175,20 +178,9 @@ def append(self, key, value=None):
 				...
 			})
 		"""
-		if value==None:
-			value={}
-		if isinstance(value, (dict, BaseDocument)):
-			if not self.__dict__.get(key):
-				self.__dict__[key] = []
-			value = self._init_child(value, key)
-			self.__dict__[key].append(value)
-
-			# reference parent document
-			value.parent_doc = self
-
-			return value
-		else:
-
+		if value is None:
+			value = {}
+		elif not isinstance(value, (dict, BaseDocument)):
```

Comment:
This behavior feels pretty inconsistent. If someone passes value as `{}`, this block executes but doesn't if someone sets it as `None` which we anyway reset to `{}` above.